cpp/src/fil/internal.cuh

```diff
@@ -520,6 +520,11 @@ void init_sparse(const raft::handle_t& h,
                  const fil_node_t* nodes,
                  const forest_params_t* params);
 
+struct predict_params;
+
+template <int NITEMS, leaf_algo_t leaf_algo, bool cols_in_shmem, class storage_type>
```

Comment:
Why is this needed both here _and_ in `common.cuh`? It definitely should be in one place only, `internal.cuh` if it is used in tests and `common.cuh` if it isn't.